woocommerce/woocommerce-ios
# [Mobile Payments] Switch from WCPayStore to PaymentGatewayAccountStore

Ready for review with the exception of needing a Order Details Data Store Unit Test update (see below)

Closes #4314 

Changes:
- [x] Add PaymentGatewayAccountResultsController to SettingsViewController so that PaymentGatewayAccounts in Storage yield instant show or no-show of "Manage Card Readers" while we wait for network response
- [x] Add `ResultsController<PaymentGatewayAccount>` to OrderDetailsResultsControllers so that PaymentGatewayAccounts in Storage yield instant show or no-show of "Collect Payment" while we wait for network response
- [x] PaymentCaptureOrchestrator to use PaymentGatewayAccountStore for collecting payments
- [x] Remove WCPayStore and WCPayAction
- [ ] Update Order Details Data Store Unit Test (will probably need a mock for PaymentGatewayAccountStore)

To Test:
```
- Delete the app from the device
- Install the app and login

- Select a store without WCPay active, or at least not in the beta
- Navigate to Settings and ensure Manage Card Reader does not appear (wait at least 15 seconds)
- Select a store with WCPay active and eligible for beta
- Ensure Manage Card Reader does appear (wait at least 15 seconds)
- Exit the app and restart it
- Navigate to Settings and ensure Manage Card Reader appears immediately
- Ensure Manage Card Reader does not disappear (wait at least 15 seconds)
- Exit the app and restart it
- Temporarily deactivate the WCPay plugin on the store
- Navigate to Settings and ensure Manage Card Reader appears immediately
- Ensure Manage Card Reader disappears within at most 15 seconds
- Reactivate the plugin on the store
- Navigate to Settings and ensure Manage Card Reader re-appears

- Deactivate the plugin
- Exit and re-enter Settings (Settings does not have a pull down to refresh at this time)
- Ensure Manage Card Readers disappears
- Re-activate the plugin
- Exit and re-enter Settings
- Ensure Manage Card Readers re-appears

- Select a store without WCPay active, or at least not in the beta
- Navigate to a cod order in processing state
- Ensure Collect Payment does not appear (wait at least 15 seconds)
- Select a store with WCPay active and eligible for beta
- Navigate to a cod order in processing state
- Ensure Collect Payment does appear (wait at least 15 seconds)
- Navigate to another cod order in processing state
- Ensure Collect Payment appears immediately

- Deactivate the plugin
- Pull down to refresh the order
- Ensure Collect Payment disappears
- Re-activate the plugin
- Pull down to refresh the order
- Ensure Collect Payment re-appears

- Complete payment collection
- Open the internal dashboard
- Ensure the statement descriptor matches what is set in the plugin's settings
```

Update release notes:
